open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/handlers/SubjectAreaCategoryHandler.java

```diff
@@ -423,4 +412,95 @@ private void updateAttributes(Category currentCategory, Category newTerm) {
 
         return response;
     }
+
+    /**
+     * Get the terms that are categorized by this Category
+     *
+     * @param userId       unique identifier for requesting user, under which the request is performed
+     * @param guid         guid of the category to get terms
+     * @param termHandler term handler
+     * @param startingFrom initial position in the stored list.
+     * @param pageSize     maximum number of definitions to return on this call.
+     * @return A list of terms is categorized by this Category
+     * when not successful the following Exception responses can occur
+     * <ul>
+     * <li> UserNotAuthorizedException           the requesting user is not authorized to issue this request.</li>
+     * <li> InvalidParameterException            one of the parameters is null or invalid.</li>
+     * <li> PropertyServerException              Property server exception. </li>
+     * </ul>
+     * */
+    public SubjectAreaOMASAPIResponse<Term> getCategorizedTerms(String userId, String guid, SubjectAreaTermHandler termHandler, int startingFrom, Integer pageSize) {
```

Comment:
Why didn't we use my code here?
<img width="806" alt="Screenshot 2020-10-20 at 23 39 26" src="https://user-images.githubusercontent.com/35165540/96641890-0ba94c80-132e-11eb-8269-f3a63f4939b6.png">

Reply:
And this method:
<img width="812" alt="Screenshot 2020-10-20 at 23 51 38" src="https://user-images.githubusercontent.com/35165540/96644407-b111ef80-1331-11eb-882f-293988c2d1c0.png">

Reply:
your task was only to add paging to these methods. But for some reason you added unnecessary checks, which are already in the Egeria Core classes.

call would look something like this:
<img width="1048" alt="Screenshot 2020-10-21 at 00 13 43" src="https://user-images.githubusercontent.com/35165540/96644937-7bb9d180-1332-11eb-8ffc-d216c452edc8.png">

Reply:
> Why didn't we use my code here?
> <img alt="Screenshot 2020-10-20 at 23 39 26" width="806" src="https://user-images.githubusercontent.com/35165540/96641890-0ba94c80-132e-11eb-8269-f3a63f4939b6.png">

I started from your code and and then extended, I lost track of what was your and what was mine! The main change I needed to make was that the mappers map between an OMRS  an entity to an OMAS concept, If the OMAS content has field that map to another entity the mapper does not map these fields. For example getting term categorizations needs to return Categories, but the parentCategory and glossary need to be obtained from other related entities.

Reply:
> your task was only to add paging to these methods. But for some reason you added unnecessary conditions and additional requests, which are already in the Egeria Core classes.
> 
> call would look something like this:
> <img alt="Screenshot 2020-10-21 at 00 13 43" width="1048" src="https://user-images.githubusercontent.com/35165540/96644937-7bb9d180-1332-11eb-8ffc-d216c452edc8.png">

Well I extended the scope to meet the requirements I needed for the UI, it made sense to address those requirements here as well. For example to be able to get the top categories.

Reply:
@davidradl 
I think all these checks are not needed here, they do not change the logic of the methods in any way. I would love to help you write this functionality, but I am very busy at my main job right now....

Reply:
@immmus Please could you detail which checks you do not think are necessary, I am happy to amend if I have checked anything that does not need checking. Many of the checks were added in response to null pointer bugs I found during development.